client/boot/index.js

```diff
@@ -161,8 +160,39 @@ function boot() {
 
 	translatorJumpstart.init();
 
-	reduxStore = createReduxStore();
+	localforage.config( {
+		name: 'calypso',
+		storeName: 'calypso_store',
+		description: 'Calypso Storage'
+	} );
+	localforage.getItem( 'redux-state', ( error, initialState ) => {
+		let reduxStore;
+		if ( error ) {
+			debug( 'failed to load initial redux-store state', error );
+			reduxStore = createReduxStore();
+		} else {
+			debug( 'loading redux-store with initial state', initialState );
+			reduxStore = createReduxStore( initialState || {} );
+		}
+
+		reduxStore.subscribe( function() {
+			let state = Object.assign( {}, reduxStore.getState() );
+			//remove dirty parts of the redux tree
+			delete state.sites;
+			delete state.themes;
```

Comment:
my point here was mainly to suggest separating out the delete-logic from the loading logic because it seems like we could end up with a mess here if more parts of the tree need to be deleted. in my recommendations above, we would simply replace those `delete` lines by this call:

```js
let state = Object.assign( {}, reduxStore.getState() );

state = pruneState( state );
```